src/core/IOUValue.js

```diff
@@ -0,0 +1,54 @@
+/* @flow */
+
+'use strict';
+
+const Value = require('./value').Value;
+const XRPValue = require('./XRPValue').XRPValue;
+const GlobalBigNumber = require('bignumber.js');
+const BigNumber = GlobalBigNumber.another({
+  ROUNDING_MODE: GlobalBigNumber.ROUND_HALF_UP,
+  DECIMAL_PLACES: 40
+});
+
+class IOUValue extends Value {
+
+  constructor(value: string | BigNumber, base: ?number) {
+    super(value, base);
+    this._type = IOUValue;
+  }
+
+  multiply(multiplicand: Value) {
+    if (multiplicand instanceof XRPValue) {
+      return super.multiply(
+        new IOUValue(
+          multiplicand._value.times(multiplicand.rippleUnits), undefined));
+    }
+    return super.multiply(multiplicand);
+  }
+
+  divide(divisor: Value) {
+    if (divisor instanceof XRPValue) {
+      return super.divide(
+        new IOUValue(divisor._value.times(divisor.rippleUnits), undefined));
+    }
+    return super.divide(divisor);
+  }
+
+  negate() {
+    return new IOUValue(this._value.neg(), undefined);
+  }
+
+  _canonicalize(value) {
+    if (value.isNaN()) {
+      throw new Error('Invalid result');
+    }
+    return new IOUValue(value.toPrecision(16), undefined);
```

Comment:
I was getting flow errors if I tried to call a function without all the variables, so I added them in. But I can add a default for base in the IOUValue constructor and that solves the flow errors.